pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go

```diff
@@ -115,10 +126,18 @@ func (cb *CollectorBundle) addCollectorFromConfig(collectorName string) {
 		return
 	}
 
+	// this is to stop multiple crds and/or people setting resources as custom resources which we already collect
+	// I am using the fullName for now on purpose in case we have the same resource across 2 different groups setup
```

Comment:
I think we don't want a resource that we already collect to be collected as CRD regardless of the version 🤔

Reply:
But we would want to support both:

 - networking.istio.io/v1alpha2/VirtualService
 - networking.istio.io/v1alpha3/VirtualService

I think in the above `isCRD` block, we should be checking the resource name against a black list (e.g. to prevent crd-style collection of pods).

Reply:
Does the notion of `preferred` API group version exists for CRD APIs like regular resources?

Reply:
Yes, i think so: https://kubernetes.io/docs/reference/using-api/#api-versioning. Especially since any resource we collect could be potentially a "normal" resource under the disguise of a crd.

Having said that, my primary idea was to keep the PR, the discovery and CRD configuration/collection in the code as simple as possible for the first iteration. 

How would you envision this to work? Right now there is a 1-1 relation between a crd set in the configuration (`datadoghq.com/datadogmetrics/v1alpha1`) and the one we will collect in the end.

Reply:
I think from small experiments I did what might be happening is the object is stored in the preferred version but both API groups are able to list it.

It could make sense to warn the user about the fact he's activating collection for a non-preferred version but at the same time he might know what he's in the process of upgrading consumers of the old version.

Let's keep that for later I guess because I'm not certain of the right move here.

Reply:
yes, let's do that in a follow-up. I don't think that should be blocking